tests/test_atom_mapping.py

```diff
@@ -953,3 +965,190 @@ def test_get_cores_and_diagnostics():
+    # we explicitly compute the ordering
+    atomic_nums = [a.GetAtomicNum() for a in mol.GetAtoms()]
+    old_to_new_mapping = {}
+    new_to_old_mapping = {}
+
+    # generate old to new mapping
+    for atom_idx, atom_num in enumerate(atomic_nums):
+        if atom_num != 1:
+            old_to_new_mapping[atom_idx] = len(old_to_new_mapping)
+
+    # generate new to old mapping
+    for old, new in old_to_new_mapping.items():
+        new_to_old_mapping[new] = old
+
+    return new_to_old_mapping
+
+
+import copy
+
+
+def test_truncated_mols():
+    mols_with_hs = read_sdf(hif2a_set, removeHs=False)
+    mols_without_hs = read_sdf(hif2a_set, removeHs=True)
+
+    n_mols = len(mols_with_hs)
+
+    with_h_visits = []
+    without_h_visits = []
+    hybrid_visits = []
+
+    with_h_core_sizes = []
+    without_h_core_sizes = []
+    hybrid_core_sizes = []
+
+    TEST_ATOM_MAPPING_KWARGS = copy.deepcopy(DEFAULT_ATOM_MAPPING_KWARGS)
+
+    # useful for testing larger cutoff settings
+    # TEST_ATOM_MAPPING_KWARGS["ring_cutoff"] = 0.4
+    # TEST_ATOM_MAPPING_KWARGS["chain_cutoff"] = 0.4
+
+    for i in range(n_mols):
+        for j in range(i + 1, n_mols):
+            print(i, j)
+            mol_a_with_h, mol_b_with_h = mols_with_hs[i], mols_with_hs[j]
+            mol_a_without_h, mol_b_without_h = mols_without_hs[i], mols_without_hs[j]
+
+            cores_h, diagnostics_with_h = atom_mapping.get_cores_and_diagnostics(
+                mol_a_with_h, mol_b_with_h, **TEST_ATOM_MAPPING_KWARGS
+            )
+            tnv_h = diagnostics_with_h.total_nodes_visited
+            core_h = cores_h[0]
+
+            cores_no_h, diagnostics_no_h = atom_mapping.get_cores_and_diagnostics(
+                mol_a_without_h, mol_b_without_h, **TEST_ATOM_MAPPING_KWARGS
+            )
+            tnv_no_h = diagnostics_no_h.total_nodes_visited
+            core_no_h = cores_no_h[0]
+
+            # hybrid method, use core from without H atom-mapping
+            ntom_mol_a = new_to_old_map_after_removing_hs(mol_a_with_h)
+            ntom_mol_b = new_to_old_map_after_removing_hs(mol_b_with_h)
+
+            core_a_initial = [ntom_mol_a[x] for x in core_no_h[:, 0]]
+            core_b_initial = [ntom_mol_b[x] for x in core_no_h[:, 1]]
+
+            initial_mapping = np.stack([core_a_initial, core_b_initial], axis=1)
+            MAPPING_KWARGS_WITH_MAPPING = copy.deepcopy(TEST_ATOM_MAPPING_KWARGS)
+            MAPPING_KWARGS_WITH_MAPPING["initial_mapping"] = initial_mapping
+
+            cores_hybrid, diagnostics_hybrid = atom_mapping.get_cores_and_diagnostics(
+                mol_a_with_h, mol_b_with_h, **MAPPING_KWARGS_WITH_MAPPING
+            )
+
+            tnv_hybrid = diagnostics_hybrid.total_nodes_visited
+            core_hybrid = cores_hybrid[0]
+
+            print("TNV: all_hs, no_hs, hybrid", tnv_h, tnv_no_h, tnv_no_h + tnv_hybrid)
+            print("CORE SIZE: all_hs, no_hs, hybrid", len(core_h), len(core_no_h), len(core_hybrid))
```

Comment:
This generates a lot of lines when running tests. Should it be commented out or behind a debug flag?

Reply:
Commented out in https://github.com/proteneer/timemachine/pull/1254/commits/c8bf1aa22edbe57e00bb9b3a58d5ad59bc0cf137